src/megatron/bridge/training/config.py

```diff
@@ -299,7 +299,6 @@ def build_datasets(self, context: DatasetBuildContext) -> Tuple[Optional[Any], O
         pass
 
 
-@dataclass
```

Comment:
could you explain why this is no longer a dataclass? what does the serialized config container yaml now look like with this change? 

why not do something like this to keep `seq_length` and `sequence_length` in sync? https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/b245756e35943083a77aff6217fe60dd1704f6ca/src/megatron/bridge/training/mixed_precision.py#L74-L80